source/lambda/customResourceKendraDataSource/lambda_function.py

```diff
@@ -20,56 +20,58 @@ def on_create(event, context):
+    # get a list of all covid pdfs and their associated kendra policy
+    response = s3_client.list_objects_v2(Bucket=dataBucketName, MaxKeys=1000)
+
+    s3_resource = boto3.resource('s3', region_name=os.environ['AWS_REGION'])
+
+    # copy over in the bulk bucket the kendra json policy files
+    for doc in response['Contents']:
+        
+        if doc['Key'].split('.')[-1] == 'json':
+            
+            copy_source = {
+                'Bucket': dataBucketName,
+                'Key': doc['Key']
             }
-        },
-        Description='Covid pdfs data',
-        RoleArn=roleArn
-    )
-    dataSourceId = create_data_source_response['Id']
-    print("Data source id {} for kendra index with id {} created.".format(dataSourceId, kendraIndexId))
-    print("syncing the s3 data source")
-    response = kendra_client.start_data_source_sync_job(
-        Id=dataSourceId,
-        IndexId=kendraIndexId
-    )
-    print("Started sync job for data source id :{}, the execution id of the sync job is :{}".format(dataSourceId,response['ExecutionId']))
+            
+            destinationKey = doc['Key'].split('/')[2]
+
+            s3_resource.meta.client.copy(copy_source,
+                                     os.environ['BULK_PROCESSING_BUCKET'],
+                                     'kendraPolicyDrop/' + destinationKey)
+
+    # copy over in the bulk bucket the pdfs
+    for doc in response['Contents']:
+        
+        if doc['Key'].split('.')[-1] == 'pdf':
+            
+            copy_source = {
+                'Bucket': dataBucketName,
+                'Key': doc['Key']
+            }
+            
+            destinationKey = doc['Key'].split('/')[2]
+
+            s3_resource.meta.client.copy(copy_source,
+                                     os.environ['BULK_PROCESSING_BUCKET'],
+                                     'documentDrop/' + destinationKey)
+
+    dataSourceId = "None"
     return {'PhysicalResourceId' : dataSourceId}
-   
+
+
 def on_update(event, context):
-    kendra_client = boto3.client('kendra')
-    kendraIndexId = os.environ['KENDRA_INDEX_ID']
-    print("Updating the s3 data source and connector for kendra index id: {}".format(kendraIndexId))
-    dataBucketName = os.environ['DATA_BUCKET_NAME']
-    roleArn = os.environ['KENDRA_ROLE_ARN']
-    dataSourceId = event['PhysicalResourceId']
-    # Name of the data source cannot be updated. Hence we do not give the customers an option to do so.
-    try:
-        update_data_source_response = kendra_client.update_data_source(
-            Id=dataSourceId,
-            Name='DUSCovidDataset',
-            IndexId= kendraIndexId,
-            Configuration={
-                'S3Configuration': {
-                    'BucketName': dataBucketName,
-                }
-            },
-            Description='Covid pdfs data',
-            RoleArn=roleArn
-        )
-        return {'PhysicalResourceId' : dataSourceId}
-    except:
-        raise
+    # not in use
```

Comment:
As of now there is not an ideal way to update the Kendra bulk processing on stackupdates.
It may lead to unexpected changes as we dont know the diff of the new documents added.
For now, lets have no actions on stack updates.